dbms/src/AggregateFunctions/AggregateFunctionSumMap.h

```diff
@@ -0,0 +1,192 @@
+    {
+        Field field_keys;
+        column_keys.get(row_num, field_keys);
+        const auto & keys = field_keys.get<Array &>();
+
+        Field field_values;
+        column_values.get(row_num, field_values);
+        const auto & values = field_values.get<Array &>();
+
+        auto & merged_maps = this->data(place).merged_maps;
+
+        if (keys.size() != values.size())
+            throw Exception("Sizes of keys and values arrays do not match", ErrorCodes::LOGICAL_ERROR);
+
+        size_t size = keys.size();
+
+        for (size_t i = 0; i < size; ++i)
+        {
+            if (merged_maps.find(keys[i]) != merged_maps.end())
+                applyVisitor(FieldVisitorSum(values[i]), merged_maps[keys[i]]);
+            else
+                merged_maps[keys[i]] = values[i];
+        }
+    }
+
+    void merge(AggregateDataPtr place, ConstAggregateDataPtr rhs, Arena * arena) const override
+    {
+        auto & merged_maps = this->data(place).merged_maps;
+        const auto & rhs_maps = this->data(rhs).merged_maps;
+
+        for (const auto &rhs_map : rhs_maps)
+        {
+            if (merged_maps.find(rhs_map.first) != merged_maps.end())
+                applyVisitor(FieldVisitorSum(rhs_map.second), merged_maps[rhs_map.first]);
+            else
+                merged_maps[rhs_map.first] = rhs_map.second;
+        }
+    }
+
+    void serialize(ConstAggregateDataPtr place, WriteBuffer & buf) const override
+    {
+        const auto & merged_maps = this->data(place).merged_maps;
+        size_t size = merged_maps.size();
+        writeVarUInt(size, buf);
+
+        for (const auto &v : merged_maps)
+        {
+            keys_type->serializeBinary(v.first, buf);
+            values_type->serializeBinary(v.second, buf);
+        }
+    }
+
+    void deserialize(AggregateDataPtr place, ReadBuffer & buf, Arena *) const override
+    {
+        auto & merged_maps = this->data(place).merged_maps;
+
+        size_t size = 0;
+        readVarUInt(size, buf);
+
+        for (size_t i = 0; i < size; ++i)
+        {
+            Field key, value;
+            keys_type->deserializeBinary(key, buf);
+            values_type->deserializeBinary(value, buf);
+            merged_maps[key] = value;
+        }
+    }
+
+    void insertResultInto(ConstAggregateDataPtr place, IColumn & to) const override
+    {
+        auto & to_cols = static_cast<ColumnTuple &>(to).getColumns();
+
+        auto & to_keys_arr = static_cast<ColumnArray &>(*to_cols[0]);
+        auto & to_values_arr = static_cast<ColumnArray &>(*to_cols[1]);
+
+        auto & to_keys_col = to_keys_arr.getData();
+        auto & to_keys_offsets = to_keys_arr.getOffsets();
+
+        auto & to_values_col = to_values_arr.getData();
+        auto & to_values_offsets = to_values_arr.getOffsets();
+
+        const auto & merged_maps = this->data(place).merged_maps;
+        size_t size = merged_maps.size();
+
+        to_keys_col.reserve(size);
+        to_values_col.reserve(size);
```

Comment:
Why we reserve size for just inserted array even if column can already have multiple array values?

Reply:
What do you mean? I do not see just inserted array. `to_keys` and `to_values` are of the type ColumnVector<T> and I thought allocating memory for vector elements before inserting them will be more efficient. Please feel free to remove this or change it to something more efficient.

Reply:
`to_keys_col` and `to_values_col` could already contain many arrays, but you reserve just for one array.